tests/test_commands.py

```diff
@@ -124,6 +125,221 @@ def setUp(self):
+        """The reactor is set via the spider settings to the asyncio value.
+
+        AsyncCrawlerProcess, the asyncio reactor, both spiders work."""
+        for spider in ["sp", "aiosp"]:
+            self._replace_custom_settings(
+                spider, f"{{'TWISTED_REACTOR': '{_asyncio_reactor_path}'}}"
+            )
+            self._assert_spider_works(self.ASYNC_MSG, spider)
+
+    def test_spider_settings_asyncio_cmdline_empty(self):
+        """The reactor is set via the spider settings to the asyncio value
+        and via command line to the empty value. The command line value takes
+        precedence so the spider settings don't matter.
+
+        CrawlerProcess, the default reactor, only the normal spider works."""
+        for spider in ["sp", "aiosp"]:
+            self._replace_custom_settings(
+                spider, f"{{'TWISTED_REACTOR': '{_asyncio_reactor_path}'}}"
+            )
+
+        self._assert_spider_works(self.NORMAL_MSG, "sp", "-s", "TWISTED_REACTOR=")
+        self._assert_spider_asyncio_fail(
+            self.NORMAL_MSG, "aiosp", "-s", "TWISTED_REACTOR="
+        )
+
+    def test_project_empty_spider_settings_asyncio(self):
+        """The reactor is set via the project settings to the empty value
+        and via the spider settings to the asyncio value. CrawlerProcess is
+        chosen based on the project settings, but the asyncio reactor is chosen
+        based on the spider settings.
+
+        CrawlerProcess, the asyncio reactor, both spiders work."""
+        self._append_settings("TWISTED_REACTOR = None\n")
+        for spider in ["sp", "aiosp"]:
+            self._replace_custom_settings(
+                spider, f"{{'TWISTED_REACTOR': '{_asyncio_reactor_path}'}}"
+            )
+            self._assert_spider_works(self.NORMAL_MSG, spider)
+
+    def test_project_asyncio_spider_settings_select(self):
+        """The reactor is set via the project settings to the asyncio value
+        and via the spider settings to the select value. AsyncCrawlerProcess
+        is chosen based on the project settings, and the conflicting reactor
+        setting in the spider settings causes an exception.
+
+        AsyncCrawlerProcess, the asyncio reactor, both spiders produce a
+        mismatched reactor exception."""
+        self._append_settings(f"TWISTED_REACTOR = '{_asyncio_reactor_path}'\n")
+        for spider in ["sp", "aiosp"]:
+            self._replace_custom_settings(
+                spider,
+                "{'TWISTED_REACTOR': 'twisted.internet.selectreactor.SelectReactor'}",
+            )
+            _, out, err = self.proc(self.name, spider)
+            assert self.ASYNC_MSG in out, out
+            assert (
+                "The installed reactor (twisted.internet.asyncioreactor.AsyncioSelectorReactor)"
+                " does not match the requested one"
+                " (twisted.internet.selectreactor.SelectReactor)"
+            ) in err, err
```

Comment:
I'll try to change this (and other cases where it's emitted) to a more specific message separately but I don't yet know if it's possible.